src/fidesops/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -132,6 +137,26 @@ def create_privacy_request(
                 privacy_request.cache_identity(identity)
                 privacy_request.cache_encryption(privacy_request_data.encryption_key)
 
+            # Store masking secrets in the cache
+            logger.info(
+                f"Caching masking secrets for privacy request {privacy_request.id}"
+            )
+            erasure_rules: List[Rule] = policy.get_rules_for_action(
+                action_type=ActionType.erasure
+            )
+            unique_masking_strategies_by_name: Set[str] = set()
+            for rule in erasure_rules:
+                unique_masking_strategies_by_name.add(rule.masking_strategy.strategy)
+            # todo- could generate all secrets for all strategies
```

Comment:
Currently, this bit of code only generates the secrets needed for a given request. This means that, in our `AES` strategy, we need to go aheads and generate secrets for `HMAC` in the `AES` context, because we need `HMAC` secrets to generate the nonce. These are the HMAC-in-AES-context secrets I'm referring to: https://github.com/ethyca/fidesops/pull/119/files#diff-3c3364723154032848cad20814d09705811bb0abbe49450ca21921e8af0468ccR9-R11

Alternatively, we could blindly generate all secrets for all strategies at this point, which means we won't need to generate separate `HMAC` secrets just for `AES`.